service/resource/namespace/resource.go

```diff
@@ -173,7 +173,7 @@ func (r *Resource) GetDeleteState(ctx context.Context, obj, currentState, desire
 }
 
 func (r *Resource) GetUpdateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, interface{}, interface{}, error) {
-	return &apiv1.Namespace{}, &apiv1.Namespace{}, &apiv1.Namespace{}, nil
+	return nil, nil, nil, nil
```

Comment:
I remember we had a discussion in the azure-operator where I said we should rather return initialized empty types instead of nil because the helper methods would return an error if the right type is not received. At the moment I see this error. 

```
{"caller":"github.com/giantswarm/kvm-operator/vendor/github.com/giantswarm/operatorkit/framework/framework.go:90","error":"[{/go/src/github.com/giantswarm/kvm-operator/vendor/github.com/giantswarm/operatorkit/framework/framework.go:271: } {/go/src/github.com/giantswarm/kvm-operator/service/resource/namespace/resource.go:200: } {Namespace \"\" is invalid: metadata.name: Required value: name or generateName is required}]","event":"update","time":"17-09-15 16:01:53.631"}
```

This happens because in `ProcessCreate` we try to add the desired namespace in case it is not nil. Trying to create the empty namespace fails though. Funny thing is the helper method already deals with nil. See https://github.com/giantswarm/kvm-operator/blob/2dabae2efb9e6932fca5cd5a14cd0a4bebc99c08/service/resource/namespace/resource.go#L257-L260. So this change causes the namespace resource to behave as expected. It is probably a good idea to make the helper methods able to deal with nil types properly and use nil as zero value.

Reply:
Successfully tested on lycan.